compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1680,12 +1681,12 @@ private void checkForExportableType(BTypeSymbol symbol, Location pos, HashSet<BT
                 checkForExportableType(invokableType.retType.tsymbol, pos, visitedSymbols);
                 return;
             case TypeTags.PARAMETERIZED_TYPE:
-                BTypeSymbol parameterizedType = ((BParameterizedType) symbol.type).paramValueType.tsymbol;
+                BTypeSymbol parameterizedType = ((BParameterizedType) symbolType).paramValueType.tsymbol;
                 checkForExportableType(parameterizedType, pos, visitedSymbols);
                 return;
             case TypeTags.ERROR:
                 if (Symbols.isFlagOn(symbol.flags, Flags.ANONYMOUS)) {
-                    checkForExportableType((((BErrorType) symbol.type).detailType.tsymbol), pos, visitedSymbols);
+                    checkForExportableType((((BErrorType) symbolType).detailType.tsymbol), pos, visitedSymbols);
```

Comment:
Wouldn't it be easier to add a case for the reference type?

Reply:
I will update that

Reply:
Since the method parameter is `BTypeSymbol` no need to get the referred type.

`private void checkForExportableType(BTypeSymbol symbol, Location pos, HashSet<BTypeSymbol> visitedSymbols)`

Reply:
But it should. Even the `tsymbol.type` can be a reference type
eg
```ballerina
type Foo int;
type Bar Foo;

public function foo() returns Bar {
  return 1;
}
```

Reply:
Updated 30af85e995258ddfb1e9ebe306d5c35c46c35054